[New Product] GraalVM (#2044)
captn3m0/endoflife.date: products/graalvm.md

---
title: GraalVM
category: lang
iconSlug: NA
permalink: /graalvm
releasePolicyLink: https://www.graalvm.org/release-notes/release-calendar/
changelogTemplate: "https://www.graalvm.org/release-notes/{{ '__RELEASE_CYCLE__' | replace:'.','_' }}/"
LTSLabel: "<abbr title='Annual Release'>AR</abbr>"
eolColumn: Maintenance
releaseColumn: true
releaseDateColumn: true

> # Disabled because tags date on https://github.com/graalvm/graalvm-ce-builds/tags are not the

Reviewer: How many/Which of these are mismatching?

Author: Almost all of them [on the first tags page](https://github.com/graalvm/graalvm-ce-builds/tags). Most only by a few days, but some are really wrong.

For example for GraalVM 22.3.0 (the last version), vm-22.3.0 tag date is 2022-07-26, but the release date is 2022-10-25. 
If you go to this version [GitHub release page](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-22.3.0) the release date is OK, but you would have to use [the GitHub release API](https://github.com/endoflife-date/release-data/issues/34) to get it.

Reviewer: Seeing as there are only gonna be quarterly updates here, manual updates should be good for now.

> It seems tags have been created a long time after the releases.

I was confused by this, which seems incorrect - the tags were created a long time _before the release_.

I did look at the [Docker Images](https://github.com/graalvm/container/pkgs/container/graalvm-ce/versions?filters%5Bversion_type%5D=tagged), but they are unreliable as well, as they seem to be overwritten.

Author: Ok, and sorry for the wrong commit message. I fixed it.

    releaseDate: 2020-05-19
    eol: 2020-08-18
    lts: false
    latest: "20.1.0"
    latestReleaseDate: 2020-05-19

-   releaseCycle: "20.0"
    releaseDate: 2020-02-18
    eol: 2020-05-19
    lts: false
    latest: "20.0.0"
    latestReleaseDate: 2020-02-18

-   releaseCycle: "19.3"
    releaseDate: 2019-11-19
    # EOL date should have been 2020-11-19, but there were two releases after that. Using latestReleaseDate.
    eol: 2021-04-20
    lts: true
    latest: "19.3.6"
    latestReleaseDate: 2021-04-20

---

> [GraalVM](https://www.graalvm.org/) is a Java VM and JDK based on HotSpot/OpenJDK, implemented in
> Java. It supports additional programming languages and execution modes, like ahead-of-time
> compilation of Java applications for fast startup and low memory footprint.

GraalVM Community release cadence is fixed. There is one feature release every three months, always
on the Tuesday closest to the 17th of the months of January, April, July, and October. Each feature
release supersedes any previous one, except annual releases.

Each year, the fourth feature release (for example 22.3) receives bugfixes for the next 12 months.
Such releases are called annual releases.

> An enterprise edition of GraalVM is available: it is called _GraalVM Enterprise_. This edition is
> maintained by Oracle and have LTS releases supported for at least two years. More information can be
> found [on the Oracle website](https://docs.oracle.com/en/graalvm/index.html).

Reviewer: Slightly briefer:

```suggestion
[_GraalVM Enterprise_](https://docs.oracle.com/en/graalvm/index.html) is maintained by Oracle, with each LTS release supported for at least two years.
```

Author: What do you think of this ?

> A commercial offering with long term support is proposed by Oracle with [GraalVM Enterprise](https://docs.oracle.com/en/graalvm/index.html).

I would like to make it clear that GraalVM Enterprise is not free.

Reviewer: s/propose/supported perhaps:

> A commercial offering with long term support is supported by Oracle as [GraalVM Enterprise](https://docs.oracle.com/en/graalvm/index.html).

Author: Having support/supported looks a bit odd. Let's use that sentence with "provided" (similar to what you suggested on https://github.com/endoflife-date/endoflife.date/pull/2033#discussion_r1049386877).

Author: Done.
